cinema/views.py

```diff
@@ -0,0 +1,47 @@
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+from rest_framework import status
+from django.http import HttpResponse
+
+from .models import Movie
+from .serializers import MovieSerializer
+
+
+@api_view(["GET", "POST"])
+def movie_list(request):
+    if request.method == "GET":
+        movies = Movie.objects.all()
+        serializer = MovieSerializer(movies, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    elif request.method == "POST":
+        serializer = MovieSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(["GET", "PUT", "DELETE"])
+def movie_detail(request, pk: int):
+    try:
+        movie_single_object = Movie.objects.get(pk=pk)
```

Comment:
Make sure you catch an error in @api_view in case that object doesn't exist. Use get_object_or_404 instead of try/except for this purpose.